### PR TITLE
various: cleanup of "inherit version;"

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
a bit related to #103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130469` at 82cc1ef6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.osmosdr</li>
  </ul>
</details>
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>pothos</li>
    <li>sdrangel</li>
  </ul>
</details>
<details>
  <summary>16 packages built successfully:</summary>
  <ul>
    <li>cubicsdr</li>
    <li>dump1090</li>
    <li>gnss-sdr</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gqrx</li>
    <li>libbladeRF</li>
    <li>maude</li>
    <li>python38Packages.soapysdr-with-plugins</li>
    <li>python39Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>rtl_433</li>
    <li>soapybladerf</li>
    <li>soapysdr-with-plugins</li>
    <li>welle-io</li>
    <li>wishbone-tool</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/libbladeRF/default.nix:64:5:
```
   |
64 |     license = licenses.lgpl21;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/maude/default.nix:20:3:
```
   |
20 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

flex is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/maude/default.nix:20:3:
```
   |
20 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

bison is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/interpreters/maude/default.nix:20:3:
```
   |
20 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/interpreters/maude/default.nix [hunk 1]
```diff
@@ -2,26 +2,20 @@
...
-  inherit version;
+  version = "3.1";
 
   src = fetchurl {
     url = "http://maude.cs.illinois.edu/w/images/d/d3/Maude-${version}.tar.gz";
     sha256 = "b112d7843f65217e3b5a9d40461698ef8dab7cbbe830af21216dfb924dc88a2f";
   };
 
+  fullMaude = fetchurl {
+    url = "http://maude.cs.illinois.edu/w/images/0/0a/Full-Maude-${version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/maude/default.nix:15]:
```suggestion
    url = "https://maude.cs.illinois.edu/w/images/d/d3/Maude-${version}.tar.gz";
    sha256 = "b112d7843f65217e3b5a9d40461698ef8dab7cbbe830af21216dfb924dc88a2f";
  };

  fullMaude = fetchurl {
    url = "https://maude.cs.illinois.edu/w/images/0/0a/Full-Maude-${version}.zip";
```

File: pkgs/development/libraries/libbladeRF/default.nix [hunk 2]
```diff
@@ -23,6 +12,14 @@ in stdenv.mkDerivation {
...
   };
 
+  # fetch submodule
+  noos = fetchFromGitHub {
+    owner = "analogdevicesinc";
+    repo = "no-OS";
+    rev = "0bba46e6f6f75785a65d425ece37d0a04daf6157";
+    sha256 = "0is79dhsyp9xmlnfdr1i5s1c22ipjafk9d35jpn5dynpvj86m99c";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libbladeRF/default.nix:22]: This should stay in the let in to not accidental turn into an env variable when being left unused.

**Comment by Mindavi** [comment]
LGTM. Seems like this now only causes 3 rebuilds, probably due to the url from maude changing from http to https.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.osmosdr</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pothos</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>cubicsdr</li>
    <li>dump1090</li>
    <li>gnss-sdr</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gqrx</li>
    <li>libbladeRF</li>
    <li>maude</li>
    <li>python38Packages.soapysdr-with-plugins</li>
    <li>python39Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>soapybladerf</li>
    <li>soapysdr-with-plugins</li>
    <li>welle-io</li>
    <li>wishbone-tool</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.osmosdr</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>pothos</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>cubicsdr</li>
    <li>dump1090</li>
    <li>gnss-sdr</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gqrx</li>
    <li>libbladeRF</li>
    <li>maude</li>
    <li>python38Packages.soapysdr-with-plugins</li>
    <li>python39Packages.soapysdr-with-plugins</li>
    <li>qradiolink</li>
    <li>rtl_433</li>
    <li>sdrangel</li>
    <li>soapybladerf</li>
    <li>soapysdr-with-plugins</li>
    <li>welle-io</li>
    <li>wishbone-tool</li>
  </ul>
</details>

File: pkgs/development/interpreters/maude/default.nix [hunk 3]
```diff
@@ -2,26 +2,20 @@
...
   src = fetchurl {
-    url = "http://maude.cs.illinois.edu/w/images/d/d3/Maude-${version}.tar.gz";
+    url = "https://maude.cs.illinois.edu/w/images/d/d3/Maude-${version}.tar.gz";
     sha256 = "b112d7843f65217e3b5a9d40461698ef8dab7cbbe830af21216dfb924dc88a2f";
   };
 
+  fullMaude = fetchurl {
+    url = "https://maude.cs.illinois.edu/w/images/0/0a/Full-Maude-${version}.zip";
+    sha256 = "8b13af02c6243116c2ef9592622ecaa06d05dbe1dd6b1e595551ff33855948f2";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/maude/default.nix:17]: Please move that back, so we avoid the rebuilds.

File: pkgs/development/tools/misc/wishbone-tool/default.nix [hunk 4]
```diff
@@ -1,19 +1,16 @@
 { lib, fetchFromGitHub, rustPlatform, libusb-compat-0_1 }:
 
-let
+rustPlatform.buildRustPackage rec {
+  pname = "wishbone-tool";
   version = "0.6.9";
-  src = fetchFromGitHub {
+  github = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/wishbone-tool/default.nix:6]: this should stay in the let in but I think we should keep the variable rename for clarity.

Comment (Stunkymonkey) [reply pkgs/development/tools/misc/wishbone-tool/default.nix:6]: i will revert the changes made to this file.

**Comment by SuperSandro2000** [comment]
@Stunkymonkey sorry but we just collected a merge conflict.

**Comment by Stunkymonkey** [comment]
no problem just rebased it and fixed it

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Pull request merged**